allennlp/data/token_indexers/pretrained_transformer_mismatched_indexer.py

```diff
@@ -64,6 +64,9 @@ def tokens_to_indices(self, tokens: List[Token], vocabulary: Vocabulary) -> Inde
         output = self._matched_indexer.tokens_to_indices(tokens, vocabulary)
 
         # self._intra_word_tokenize() does not insert special tokens, so we need to do it here
+        output["type_ids"] = self._tokenizer.create_token_type_ids_from_sequences(
```

Comment:
Perhaps we should move this behind a flag? I'm not sure if people always want type ids. Will take up extra space, for instance.

Reply:
This is how type ids are handled inside the huggingface library:
```Python
    def forward(self, input_ids=None, token_type_ids=None, position_ids=None, inputs_embeds=None):
        if token_type_ids is None:
            token_type_ids = torch.zeros(input_shape, dtype=torch.long, device=device)
```
I can let them add the zero tensor instead of doing it here, but then I just have to drag around a special case.

The other reason we should have them is that there might be a transformer that uses type ids to mark special tokens. I don't know of any, but there are many in the library that I'm not familiar with.

Reply:
Sounds good.